src/Mvc/Mvc.Core/src/ModelBinding/Binders/DictionaryModelBinder.cs

```diff
@@ -142,7 +142,7 @@ public override async Task BindModelAsync(ModelBindingContext bindingContext)
         // Attempt to bind dictionary from a set of prefix[key]=value entries. Get the short and long keys first.
         var prefix = bindingContext.ModelName;
         var keys = enumerableValueProvider.GetKeysFromPrefix(prefix);
-        if (keys.Count == 0)
+        if (string.IsNullOrEmpty(prefix) || keys.Count == 0)
```

Comment:
Based on the lines below, this is a last ditch effort to find prefix[key] and return a validation error only if BindingRequired. As such, if prefix is null, it should follow the same AddErrorIfBindingRequired logic.